### PR TITLE
Flush logs to HTTP connection after each write

### PR DESCRIPTION
**Pull request by titanous**
This fixes tailing logs.

@lmars for review.

### CODE REVIEW

**Comment by titanous** [comment]
Hmm, I should probably add a test for this...

File: jobs.go [hunk 1]
```diff
@@ -118,14 +118,31 @@ func jobLog(req *http.Request, app *ct.App, params martini.Params, cluster clust
...
+				nw, ew := w.Write(buf[:nr])
+				if ew != nil || nr != nw {
+					return
+				}
+				w.(http.Flusher).Flush()
+			}
+			if er != nil {
+				return
+			}
+		}
```
Comment (lmars) [inline jobs.go]: Would it be simpler to wrap `w` in a `FlushWriter` which would flush after every write?

``` go
type FlushWriter struct {
        w io.Writer
}

func (f *FlushWriter) Write(p []byte) (int, error) {
        defer func() {
                if _, ok := f.w.(http.Flusher); ok {
                        f.w.Flush()
                }
        }()
        return f.w.Write(p)
}
```

**Comment by titanous** [comment]
@lmars Added tests and refactored a bunch.

**Comment by lmars** [comment]
LGTM

**Pull request merged**